### PR TITLE
Modularized with openapi

### PR DESCRIPTION
**Pull request by ghost**
Modularisert komponenten, så at OpenAPI-specen er i en modul, og selve applikasjonen i en.

Der er det gamle rest-interfacet parallellt med det nya. Jeg har også kopiert servicerne og ryddet de opp litt, så der er litt kodeduplisering. Tanken min er at vi kan fjerne det gamle interfacet og de gamle servicene samtidlig når tid er.

**Ved kodelesing: Det er lurt at se på hver commit for seg!**

### CODE REVIEW

File: api/src/main/resources/soknadsfillager-api.yml [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+          schema:
+            type: string
+
+      responses:
+        200:
+          description: The requested files were deleted.
+      tags:
+        - files
+
+components:
```
Comment (alpet) [inline api/src/main/resources/soknadsfillager-api.yml:157]: I would have split this pull request in 3:
1. Your open API configuration
2. Your model
3. your API.
I dont like spending so much time reviewing.

File: fillager/src/main/kotlin/no/nav/soknad/arkivering/soknadsfillager/service/GetFilesService.kt [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+import no.nav.soknad.arkivering.soknadsfillager.supervision.Operations
+import org.slf4j.LoggerFactory
+import org.springframework.stereotype.Service
+import java.time.OffsetDateTime
+import java.time.ZoneOffset
+
+@Service
+class GetFilesService(private val filRepository: FilRepository, private val fileMetrics: FileMetrics) {
+	private val logger = LoggerFactory.getLogger(javaClass)
+
```
Comment (alpet) [inline fillager/src/main/kotlin/no/nav/soknad/arkivering/soknadsfillager/service/GetFilesService.kt:19]: Are you refactoring in the same pull request as the API changes?

File: api/pom.xml [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+		<jackson-databind-nullable.version>0.2.2</jackson-databind-nullable.version>
+
+		<kotlin.version>1.5.20</kotlin.version>
+		<java.version>11</java.version>
+	</properties>
+
+	<dependencies>
+		<dependency>
+			<groupId>org.springframework</groupId>
+			<artifactId>spring-web</artifactId>
```
Comment (alpet) [inline api/pom.xml:31]: Why the API package needs spring-web dependency ? I thought the controller is in the other package ?

**Review by alpet — APPROVED** [review]
The pull request is not built in code reviewer friendly way. Otherwise all is good.

File: api/pom.xml (hunk 3 above)
Comment (ghost) [reply api/pom.xml:31]: The generated code won't build without the dependency.

File: api/src/main/resources/soknadsfillager-api.yml (hunk 1 above)
Comment (ghost) [reply api/src/main/resources/soknadsfillager-api.yml:157]: Yes, very reasonable. I'll try to keep it smaller. Thanks for reviewing. 👍

**Pull request merged**